library/alloc/src/vec.rs

```diff
@@ -2730,6 +2730,26 @@ impl<T> Iterator for IntoIter<T> {
     fn count(self) -> usize {
         self.len()
     }
+
+    #[inline]
+    fn last(self) -> Option<T> {
+        unsafe {
+            if self.ptr as *const _ == self.end {
+                None
+            } else {
+                if mem::size_of::<T>() == 0 {
+                    /// Immediately marches to end of the iterator.
+                    self.ptr = self.end;
+                    // Make up a value of this ZST.
+                    Some(mem::zeroed())
+                } else {
+                    /// Immediately marches to end of the iterator.
+                    self.ptr = self.end;
+                    Some(ptr::read(self.ptr.offset(-1)))
+                }
```

Comment:
```suggestion
                /// Immediately marches to end of the iterator.
                self.ptr = self.end;
                Some(if mem::size_of::<T>() == 0 {
                    // Make up a value of this ZST.
                    mem::zeroed()
                } else {
                    ptr::read(self.ptr.offset(-1))
                })
```
Reduce duplicate logic